Chess/desk.h

```diff
@@ -1,25 +1,36 @@
 #ifndef DESK_H
 #define DESK_H
 
-#include <QObject>
+#include "common_types.h"
+#include "figure.h"
+
 #include <vector>
 
+
 class Desk : public QObject
 {
     Q_OBJECT
 public:
+    friend class Command;
+
     explicit Desk(QObject *parent = 0);
 
     void fillDefault();
+    void clear();
 
-signals:
-
-public slots:
+    Figure *getFigure(const Cell &) const;
+    bool inRange(const Cell &) const;
 
 private:
     QObject *owner_;    //ptr to desk owner (Game instance)
-    std::vector< std::vector<QObject*> > buffer_;
+    std::vector< std::vector<Figure*> > buffer_;

```

Comment:
YES! Figure* instesd of QObject* That's what I like!
Bad news - thay are row pointers! I would use shared_ptr<Figure>